### PR TITLE
feat(cli): Create-overview determine max zoom overview from the gsd

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/cli/src/cli/overview/action.create.overview.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { JobTiles, tile } from './tile.generator.js';
 import { SimpleTimer } from './timer.js';
 
 const DefaultMaxZoom = 15;
+const MaxNumberTiles = 250000;
```
Comment (blacha) [inline packages/cli/src/cli/overview/action.create.overview.ts:21]: could we add a sentence explaining what these numbers do?

File: packages/cli/src/cli/overview/action.create.overview.ts [hunk 2]
```diff
@@ -115,6 +114,7 @@ export class CommandCreateOverview extends CommandLineAction {
         qk = QuadKey.parent(qk);
       }
     }
+    if (tiles.size > MaxNumberTiles) this.prepareTiles(files, maxZoom - 1);
```
Comment (blacha) [inline packages/cli/src/cli/overview/action.create.overview.ts:117]: return?

Comment (blacha) [reply packages/cli/src/cli/overview/action.create.overview.ts:117]: also why is this async?

File: packages/cli/src/cli/overview/action.create.overview.ts [hunk 3]
```diff
@@ -126,8 +126,7 @@ export class CommandCreateOverview extends CommandLineAction {
     }
   }
 
-  async getTileMatrix(sources: ChunkSource[]): Promise<TileMatrixSet> {
-    const tiff = await CogTiff.create(sources[0]);
+  async getTileMatrix(tiff: CogTiff): Promise<TileMatrixSet> {
```
Comment (blacha) [inline packages/cli/src/cli/overview/action.create.overview.ts:129]: this doesnt need to be async the tiff is `tiff.init(true)` before.

(you are initing the tiff twice)

File: packages/cli/src/cli/overview/action.create.overview.ts [hunk 4]
```diff
@@ -55,20 +51,23 @@ export class CommandCreateOverview extends CommandLineAction {
...
     const st = new SimpleTimer();
     logger.debug({ source }, 'CreateOverview:ListTiffs');
     const tiffList = (await fsa.toArray(fsa.list(source))).filter(filterTiff);
     const tiffSource = tiffList.map((path: string) => fsa.source(path));
+
     logger.info({ source, duration: st.tick() }, 'CreateOverview:ListTiffs:Done');
 
     logger.debug({ source }, 'CreateOverview:PrepareSourceFiles');
-    const tileMatrix = await this.getTileMatrix(tiffSource);
+    const tiff = await CogTiff.create(tiffSource[0]);
```
[the remark this answers is not in the document]
Comment (Wentao-Kuang) [reply packages/cli/src/cli/overview/action.create.overview.ts:68]: We have a `filterTiff` for the file lists. So I am ok if the script just die here if the not tiff?

[the remark this answers is not in the document]
Comment (blacha) [reply packages/cli/src/cli/overview/action.create.overview.ts:68]: we should let it  die here, but with a nice error message, stack traces are not super helpful to users

File: packages/cli/src/cli/overview/action.create.overview.ts (hunk 3 above)
Comment (Wentao-Kuang) [reply packages/cli/src/cli/overview/action.create.overview.ts:129]: I need this to run `await tiff.getImage(0).loadGeoTiffTags();`, before getting the espg. The just tiff.init now working. 
image

Comment (blacha) [reply packages/cli/src/cli/overview/action.create.overview.ts:129]: The tiff.init(true) is called in the other function you should pull that out to the same place as the new CogTiff

**Review by blacha — APPROVED** [review]

**Pull request merged**